### PR TITLE
feat: enclave manager has an endpoint to create a webhook

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
This endpoint only works if a GitHub auth token is set and the enclave manager detects a `_prevenv_web_token`

### CODE REVIEW

**Comment by h4ck3rk3y** [comment]
https://github.com/kurtosis-tech/kurtosis/assets/6670312/5ffef9e5-5932-49ff-ae29-2c198726d39d

File: container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/consts/consts.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ const (
 	HttpApplicationProtocol                             = "http"
 
 	GitHubAuthStorageDirPath = "/kurtosis-data/github-auth/"
+	GithubAuthStorageToken   = "token.txt"
```
Comment (tedim52) [inline container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/consts/consts.go:36]: `s/GithubAuthStorageToken/GithubAuthTokenFileName` 

seperate - out of scope of this PR and not related to your change but just calling out- at this point LOTS of the codebase relies on  the hardcoded`/kurtosis-data` directory path. I believe it's originally set as env var to the APIC, but somehow the directory path needs to be refactored and propagated to the elements that need it.

Comment (h4ck3rk3y) [reply container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/consts/consts.go:36]: Agree with you! I just followed the current program to get something out but we should refactor ( when we have the time and have earned the right too)

File: container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/engine_functions/github_auth_storage_creator/github_auth_storage_creator.go [hunk 2]
```diff
@@ -95,6 +96,15 @@ func (creator *GitHubAuthStorageCreator) CreateGitHubAuthStorage(
 	return nil
 }
 
+// GetBestEffortGitHubAuthToken Returns empty string if no token found in [githubAuthTokenFile] or [githubAuthTokenFile] doesn't exist
+func GetBestEffortGitHubAuthToken() string {
```
Comment (tedim52) [inline container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/engine_functions/github_auth_storage_creator/github_auth_storage_creator.go]: nit: maybe? `s/GetBestEffort/Get`, I think if it's documented in the function spec/comment that behavior is empty string when no token, we don't need to add `BestEffort`. will leave up to you on this

**Review by tedim52 — APPROVED** [review]

**Pull request merged**